File: xssfinder-executor-java/src/main/java/org/xssfinder/runner/LabelledXssGeneratorImpl.java
package org.xssfinder.runner;

import org.xssfinder.LabelledXssGenerator;
import org.xssfinder.xss.XssAttack;
import org.xssfinder.xss.XssGenerator;

import java.util.HashMap;
import java.util.Map;

public class LabelledXssGeneratorImpl implements LabelledXssGenerator {
    private final XssGenerator xssGenerator;
    private final Map<String, String> labelsToAttackIds;

    public LabelledXssGeneratorImpl(XssGenerator xssGenerator) {
        this.xssGenerator = xssGenerator;
        this.labelsToAttackIds = new HashMap<String, String>();
    }

    @Override
    public String getXssAttackTextForLabel(String label) {
        XssAttack attack = xssGenerator.createXssAttack();
        labelsToAttackIds.put(label, attack.getIdentifier());
        return attack.getAttackString();
    }

    public Map<String, String> getLabelsToAttackIds() {
        return labelsToAttackIds;
    }
}

File: xssfinder-php-executor/src/XssFinder/Runner/LabelledXssGeneratorImpl.php
<?php

namespace XssFinder\Runner;

use XssFinder\Annotations\LabelledXssGenerator;
use XssFinder\Xss\XssGenerator;

class LabelledXssGeneratorImpl implements LabelledXssGenerator
{
    /** @var XssGenerator */
    private $_xssGenerator;
    private $_labelsToXssIds;

    function __construct(XssGenerator $xssGenerator)
    {
        $this->_xssGenerator = $xssGenerator;
        $this->_labelsToXssIds = array();
    }

    /**
     * @param string $label
     * @return string An XSS attack string
     */
    function getXssAttackTextForLabel($label)
    {
        $attack = $this->_xssGenerator->createXssAttack();
        $this->_labelsToXssIds[$label] = $attack->getIdentifier();
        return $attack->getAttackString();
    }

    public function getLabelsToAttackIds()
    {
        return $this->_labelsToXssIds;
    }
}